dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -2872,6 +2933,49 @@ Sedp::Writer::set_header_fields(DCPS::DataSampleHeader& dsh,
   dsh.source_timestamp_nanosec_ = now.usec() * 1000;
 }
 
+#ifdef OPENDDS_SECURITY
+Sedp::RepeatOnceWriter::RepeatOnceWriter(const RepoId& pub_id, Sedp& sedp)
+  : Writer(pub_id, sedp)
+{
+}
+
+void
+Sedp::RepeatOnceWriter::acknowledged_by_reader(const RepoId& rdr)
+{
+  RepoId remote_participant;
+  RTPS::assign(remote_participant.guidPrefix, rdr.guidPrefix);
+  remote_participant.entityId = ENTITYID_PARTICIPANT;
+
+  if (!std::memcmp(repo_id_.guidPrefix, rdr.guidPrefix, sizeof(DCPS::GuidPrefix_t))) {
```

Comment:
Should you move the ACE_Guard statement up here to protect repo_id_?

Reply:
Oh nevermind, it looks like repo_id_ is coming from TransportClient?

Reply:
Some combination of TransportClient and Sedp::Endpoint.  Not sure if we really need both.  In any case it's logically const (although not actually const).